assets/javascript/spinner/init.js

```diff
@@ -0,0 +1,177 @@
+  };
+
+  reflectError = () => {
+    window.location.href = "/ipv/page/pyi-technical";
+  };
+
+  reflectLongWait = () => {
+    this.state.spinnerState = "longWait";
+  };
+
+  initialiseTimers = () => {
+    if (this.domRequirementsMet) {
+      this.timers.informUserWhereWaitIsLong = setTimeout(() => {
+        this.reflectLongWait();
+      }, this.config.msBeforeInformingOfLongWait);
+
+      this.timers.abortUnresponsiveRequest = setTimeout(() => {
+        this.reflectError();
+      }, this.config.msBeforeAbort);
+    }
+  };
+
+  initialiseState = () => {
+    if (this.domRequirementsMet) {
+      this.state = {
+        spinnerState: "pending",
+        done: false,
+        error: false,
+        virtualDom: [],
+      };
+      const button = document.getElementById("submitButton");
+      button.setAttribute("disabled", true);
+      this.timers = {};
+    }
+  };
+
+  initialiseContent = (element) => {
+    try {
+      this.content = {
+        pending: {
+          text: element.dataset.initialSpinnerstatetext,
+          className: element.dataset.initialSpinnerstate,
+        },
+        complete: {
+          text: element.dataset.completeSpinnerstatetext || "",
+          className: element.dataset.completeSpinnerstate,
+        },
+        longWait: {
+          text: element.dataset.longwaitSpinnerstatetext,
+          className: "spinner__long-wait",
+        },
+      };
+
+      this.config = {
+        apiUrl: element.dataset.apiUrl || this.config.apiUrl,
+        msBeforeInformingOfLongWait:
+          parseInt(element.dataset.msBeforeInformingOfLongWait) ||
+          this.config.msBeforeInformingOfLongWait,
+        msBeforeAbort:
+          parseInt(element.dataset.msBeforeAbort) || this.config.msBeforeAbort,
+        msBetweenRequests:
+          parseInt(element.dataset.msBetweenRequests) ||
+          this.config.msBetweenRequests,
+      };
+
+      this.domRequirementsMet = true;
+    } catch (e) {
+      this.domRequirementsMet = false;
+    }
+  };
+
+  createVirtualDom = () => {
+    const stateContent = this.content[this.state.spinnerState];
+    const className = stateContent.className || "";
+    return [
+      {
+        nodeName: "div",
+        id: "spinner",
+        classes: ["spinner", "centre", className],
+      },
+      {
+        nodeName: "p",
+        text: stateContent.text || "",
+        classes: ["centre", "spinner-state-text", "govuk-body"],
+      },
+    ];
+  };
+
+  ifSpinnerStateChanged = (currentVDom, nextVDom) => {
+    return JSON.stringify(currentVDom) !== JSON.stringify(nextVDom);
+  };
+
+  convert = (node) => {
+    const el = document.createElement(node.nodeName);
+    if (node.text) el.textContent = node.text;
+    if (node.id) el.id = node.id;
+    if (node.classes) el.classList.add(...node.classes);
+    return el;
+  };
+
+  updateDom = () => {
+    const vDomChanged = this.ifSpinnerStateChanged(
+      this.state.virtualDom,
+      this.createVirtualDom()
+    );
```

Comment:
```suggestion
    const spinnerStateChanged = this.displayedSpinnerState !== this.spinnerState;
```

Then can remove `ifSpinnerStateChanged`